r/R/examples/stat_large.R

```diff
@@ -0,0 +1,36 @@
+#source("../core.R")
+#source("../ui.R")
+library(h2owave)
+library(tokenizers)
+base_url <- "https://programminghistorian.org/assets/basic-text-processing-in-r"
+url <- sprintf("%s/sotu_text/236.txt", base_url)
+text <- paste(readLines(url), collapse = "\n")
+sentences <- tokenize_sentences(text)
+captions <- sentences[[1]][sample(seq(1,length(sentences[[1]])),1)]
+
+card_title <- sample(row.names(mtcars),1)
+card_value <- as.character(sample(mtcars[['mpg']],1))
+card_pc <- as.integer(runif(1,1,100))
+
+test_page <- page("/page_demo")
+test_page <- page.add(test_page,"/page_demo","test",ui_large_stat_card(box="1 1 2 2"
+                                                                       ,title=card_title
+                                                                       ,value='=${{intl foo minimum_fraction_digits=2 maximum_fraction_digits=2}}'
+                                                                       ,aux_value='={{intl bar style="percent" minimum_fraction_digits=2 maximum_fraction_digits=2}}'
+                                                                       ,data=list(foo=card_value,bar=card_pc/100)
+                                                                       ,caption=captions
+                                                                       ))
+
+page.save(test_page,"/page_demo")
+
+while(TRUE)
+{
+  Sys.sleep(3)
+  card_pc <- as.integer(runif(1,1,100))
+  card_value <- as.character(sample(mtcars[['mpg']],1))
+  captions <- sentences[[1]][sample(seq(1,length(sentences[[1]])),1)]
+  test_page[[1]]$value$data$foo <- card_value
+  test_page[[1]]$value$data$bar <- card_pc/100
+  test_page[[1]]$value$caption <- captions
+  page.save(test_page,"/page_demo")
```

Comment:
Having to call `page.save(page, url)` is repetitive. 

It would be good to do something like:

```
my_page = get page '/foo' from site
...
...
save my_page
```

The above pseudo-code might look like:
```
my_page = get_page(site, '/foo')
...
save_page(my_page)
```

or

```
my_page = site$get_page('/foo')
...
my_page$save()
```

I don't know what's idiomatic in R.

Reply:
```
page <- Site("/demo")

columns = 12
rows = 10

for(column in 1:columns){
        for(row in 1:rows){
               box = paste0(column," ",row," 1 1") 
        page$add.card(paste0("card_",column,"_",row),
                              ui_markdown_card(box=box,title=box,content=''))
        }}  
page$page.save()
```

The Rdriver now is able to access the page canonically, and is able to `save` the page by `page$page.save()` (the `page` is the canonical object)